packages/drawer/src/popup.ts

```diff
@@ -0,0 +1,228 @@
+import PopupManager from '@element-plus/utils/popup-manager'
+import isServer from '@element-plus/utils/isServer'
+import {
+  reactive,
+  watch,
+  nextTick,
+  ComponentPublicInstance,
+  onBeforeUnmount,
+  onBeforeMount,
+  onMounted,
+} from 'vue'
+import {
+  getStyle,
+  addClass,
+  removeClass,
+  hasClass,
+} from '@element-plus/utils/dom'
+import { merge } from 'lodash'
+import getScrollBarWidth from '@element-plus/utils/scrollbar-width'
+
+let idSeed = 1
+let scrollBarWidth: number
+
+// XXX this may be replaced with other usePopup as Dialog?
+const usePopup = ({ props, $el, vVm }) => {
+  let _popupId: string
+  let _opening = false
+  let _closing = false
+  // let _closeTimer = null // XXX These code will be removed ？
+  // let _openTimer = null
+
+  const data = reactive({
```

Comment:
I think the name `states` or `drawerState` is better than `data`

Reply:
ok